tchannel/tchannel.py

```diff
@@ -0,0 +1,42 @@
+from __future__ import (
+    absolute_import, division, print_function, unicode_literals
+)
+
+from tornado import gen
+
+from .glossary import DEFAULT_TIMEOUT
+from .tornado import TChannel as NodeInspiredTChannel
+
+__all__ = ['TChannel']
+
+
+class TChannel(NodeInspiredTChannel):
+
+    @gen.coroutine
+    def call(self, argscheme, service, endpoint, body, headers=None, timeout=None):
+
+        if headers is None:
+            headers = {}
+
+        if timeout is None:
+            timeout = DEFAULT_TIMEOUT
+
+    @gen.coroutine
+    def call_raw(self, service, endpoint, body, headers=None, timeout=None):
+
+        response = yield self.call('raw', service, endpoint, body, headers, timeout)
+
+        raise gen.Return(response)
+
+    @gen.coroutine
```

Comment:
I would think it is too early to add coroutine decorator here. If you don't do yield inside the function, and delegate the coroutine work to some internal method. Here just return a future.